### PR TITLE
fix: handle removed float_precision kwarg in protobuf >= 7.34.0

### LINKED ISSUE #6435: feature_server.py crashes with TypeError: MessageToDict() got an unexpected keyword argument 'float_precision' on protobuf >= 7.34.0
## Summary

The feature server crashes on every `/get-online-features` and `/retrieve-online-documents` request 
when running with `protobuf >= 7.34.0`. The `float_precision` parameter was deprecated in protobuf 
6.32.x and **removed** in 7.34.0. Since Feast's `pyproject.toml` allows `protobuf >= 4.24.0`, 
fresh installs (including the official `quay.io/feastdev/feature-server:latest` image) pull 
protobuf 7.34.1, which does not accept this argument.

## Error

`TypeError: MessageToDict() got an unexpected keyword argument 'float_precision'`

Full traceback from Gunicorn worker:

`File "/opt/app-root/lib64/python3.12/site-packages/feast/feature_server.py", line 397 response_dict = await run_in_threadpool( MessageToDict, response.proto, preserving_proto_field_name=True, float_precision=18, # <-- removed in protobuf 7.34.0 )`

The worker process crashes, Gunicorn master respawns it, and the next request crashes again — 
making the online feature server completely non-functional.

## Affected Code

`sdk/python/feast/feature_server.py` — two call sites:

1. **`get_online_features` endpoint** (line ~398–401)
2. **`retrieve_online_documents` endpoint** (line ~440–443)

Both pass `float_precision=18` to `google.protobuf.json_format.MessageToDict`.

## Environment

- **Feast version:** 0.63.0
- **Image:** `quay.io/feastdev/feature-server:latest` (also `quay.io/feastdev-ci/feature-server:develop`)
- **Python:** 3.12.12
- **protobuf:** 7.34.1 (installed by pip as allowed by `protobuf>=4.24.0` in pyproject.toml)

## Reproducer

```bash
# 1. Install feast with latest protobuf
pip install feast

# 2. Verify protobuf version
python -c "import google.protobuf; print(google.protobuf.__version__)"
# Output: 7.34.1

# 3. Reproduce the crash directly
python -c "
from google.protobuf.json_format import MessageToDict
from feast.protos.feast.serving.ServingService_pb2 import GetOnlineFeaturesResponse
resp = GetOnlineFeaturesResponse()
MessageToDict(resp, preserving_proto_field_name=True, float_precision=18)
"

# Output:
# TypeError: MessageToDict() got an unexpected keyword argument 'float_precision'

```

Or with a running feature server, Any POST to the online server will crash the worker:

```
curl -X POST https://<feast-server>/get-online-features \
  -H "Content-Type: application/json" \
  -d '{"features": ["my_fv:my_feature"], "entities": {"entity_id": [1]}}'
# Returns: 500 Internal Server Error

```

### PR DESCRIPTION
**Pull request by thakoreh**
## Summary

Fixes #6435

The `float_precision` parameter was deprecated in protobuf 6.32.x and removed in 7.34.0. Since Feast allows `protobuf >= 4.24.0`, fresh installs pull protobuf 7.34.1 which crashes the feature server on every `/get-online-features` and `/retrieve-online-documents` request.

## Changes

- Added `_message_to_dict()` wrapper that conditionally passes `float_precision=18` only on protobuf < 7.34
- Replaced both `MessageToDict` call sites in `feature_server.py` with the wrapper
- Added 6 unit tests covering version detection logic and wrapper behavior

## Testing

```
6 passed in 0.05s
```

All tests verify:
- Version tuple parsing for boundary cases (7.33.x → True, 7.34.0 → False)
- Wrapper correctly includes `float_precision` on old protobuf
- Wrapper correctly omits `float_precision` on new protobuf
- Explicit `float_precision` kwarg is never overridden

## Reproducer (before fix)

```python
from google.protobuf.json_format import MessageToDict
# With protobuf >= 7.34.0:
MessageToDict(proto, preserving_proto_field_name=True, float_precision=18)
# TypeError: MessageToDict() got an unexpected keyword argument "float_precision"
```

---

<a href="https://app.devin.ai/review/feast-dev/feast/pull/6436" target="_blank">
  <picture>
    <source media="(prefers-color-scheme: dark)" srcset="https://static.devin.ai/assets/gh-open-in-devin-review-dark.svg?v=1">
    Open in Devin Review
  </picture>
</a>

### CODE REVIEW

File: sdk/python/feast/feature_server.py [hunk 1]
```diff
@@ -41,6 +41,19 @@
 from fastapi.responses import JSONResponse
 from fastapi.staticfiles import StaticFiles
 from google.protobuf.json_format import MessageToDict
+
+import google.protobuf
+
+# float_precision was removed in protobuf 7.34.0
+_PROTOBUF_VERSION = tuple(int(x) for x in google.protobuf.__version__.split(".")[:2])
+_FLOAT_PRECISION_SUPPORTED = _PROTOBUF_VERSION < (7, 34)
```
Comment (ntkathole) [inline sdk/python/feast/feature_server.py:49]: instead check if float_precision exists in MessageToDict parameters first using inspect

**Comment by thakoreh** [comment]
Closing this while I pause open-source contributions. Apologies for the noise.
